### PR TITLE
Feature/redux 2

### PR DESCRIPTION
**Pull request by sasurau4**
## Overview
Typescriptification of phenyl-redux

I compromised migration some codes by `@ts-ignore`.

### CODE REVIEW

**Comment by sasurau4** [comment]
@Naturalclar I finished migration, please review 🙏

File: modules/redux/src/phenyl-redux-module.ts [hunk 1]
```diff
@@ -64,39 +56,50 @@ export class PhenylReduxModule<TM: TypeMap> {
...
+          ...state,
+          ...action.payload,
+        }
 
       case 'phenyl/reset':
         return this.createInitialState()
 
       case 'phenyl/assign':
-        return assign(state, ...action.payload)
+        return update({ ...state, ...action.payload })
```
Comment (Naturalclar) [inline modules/redux/src/phenyl-redux-module.ts]: should be 
```
update(state, ...action.payload)
```

File: modules/redux/src/phenyl-redux-module.ts [hunk 2]
```diff
@@ -169,34 +191,43 @@ export class PhenylReduxModule<TM: TypeMap> {
...
   }
 
-  static push<N: EntityNameOf<TM>>(payload: PushActionPayload<N>): PushAction<N> {
+  static push<M extends GeneralReqResEntityMap, EN extends Key<M>>(
+    payload: PushActionPayload<EN>,
+  ): PushAction<EN> {
     return {
       type: 'phenyl/push',
-      payload: assign({ until: -1 }, payload),
+      payload: update({ until: -1, ...payload }),
```
Comment (Naturalclar) [inline modules/redux/src/phenyl-redux-module.ts]:
```
update({ until: -1} , payload)
```

File: modules/redux/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "LICENSE",
+    "README.md"
+  ],
+  "author": "Shin Suzuki <shinout310@gmail.com>",
+  "contributors": [
+    "Jesse Katsumata <jesse.katsumata@gmail.com>",
+    "Daiki Ihara <sasurau4@gmail.com>"
+  ],
+  "repository": "https://github.com/phenyl-js/phenyl/tree/master/modules/phenyl-redux",
+  "license": "Apache License 2.0",
```
Comment (Naturalclar) [inline modules/redux/package.json]: invalid value for license field. please change to `Apache-2.0`

**Pull request merged**